### PR TITLE
[tests] upload coverage

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR uploads coverage to https://codecov.io.

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+coverage:
+  status:
+    project: off
+    patch: off
+
+ignore:
+  - "tests/*"
+  - "third_party/*"
```
Comment (bukepo) [inline .codecov.yml]: should use third_party/**/*

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
-if (CMAKE_BUILD_TYPE AND OT_COMMISSIONER_COVERAGE AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+if (CMAKE_BUILD_TYPE AND OT_COMM_COVERAGE AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
```
Comment (bukepo) [inline tests/CMakeLists.txt]: OT_COVERAGE?

Comment (wgtdkp) [reply tests/CMakeLists.txt]: Should we use `OT_COMM_` as prefix of all CMake options in OT Commissioner? There will be name conflict if a project includes both `openthread` and `OT Commissioner` as a library.

Comment (bukepo) [reply tests/CMakeLists.txt]: I assume these are common flags. Just like DEBUG. User don't have to set OT_DEBUG and OT_COMM_DEBUG. Thoughts?

Comment (wgtdkp) [reply tests/CMakeLists.txt]: No, this has an assumption on libraries that they using the same options prefix. I believe it is not the usual case for CMake users. It is always for CMake user to explicitly config each dependency independently. Agree?

**Comment by jwhui** [comment]
@wgtdkp, the codecov.io upload does not seem to be happening.

**Comment by wgtdkp** [comment]
@jwhui It seems the travis build is fetching wrong commit https://travis-ci.org/openthread/ot-commissioner/builds/652305327?utm_medium=notification&utm_source=github_status. Investigating.

**Comment by wgtdkp** [comment]
This seems to be a [bug of travis](https://travis-ci.community/t/build-checks-out-wrong-pull-request-79-instead-of-7/6118). Will close this PR and create a new one.

**Pull request closed without merging**